integration-test/src/main/java/com/sequenceiq/it/cloudbreak/actor/CloudbreakUserCache.java

```diff
@@ -1,66 +1,152 @@
+        setRealUmsUserAccount(accountKey);
+
+        if (usersByAccount == null || !initializedKeys.containsKey(environmentKey) || !initializedKeys.containsValue(accountKey)) {
+            initUsers(environmentKey, accountKey);
         }
-        return usersByAccount.values().stream().flatMap(Collection::stream)
-                .filter(u -> u.getDisplayName().equals(name)).findFirst().get();
+        CloudbreakUser user = usersByAccount.values().stream().flatMap(Collection::stream)
+                .filter(u -> u.getDisplayName().equals(name)).findFirst()
+                .orElseThrow(() -> new TestFailException(String.format("There is no real UMS user with name: %s in account: %s", name, accountKey)));
+        LOGGER.info(" Real UMS user has been found in repository:: \nDisplay name: {} \nCrn: {} \nAccess key: {} \nSecret key: {} \nAdmin: {} ",
+                user.getDisplayName(), user.getCrn(), user.getAccessKey(), user.getSecretKey(), user.getAdmin());
+        return user;
+    }
+
+    public void setUsersByAccount(Map<String, List<CloudbreakUser>> users) {
+        this.usersByAccount = users;
+    }
+
+    public Map<String, List<CloudbreakUser>> getUsersByAccount() {
+        return usersByAccount;
     }
 
-    public void initUsers() {
+    public String getRealUmsUserAccount() {
+        return realUmsUserAccount;
+    }
+
+    public void setRealUmsUserAccount(String realUmsUserAccountKey) {
+        this.realUmsUserAccount = realUmsUserAccountKey;
+    }
+
+    public String getRealUmsUserEnvironment() {
+        return realUmsUserEnvironment;
+    }
+
+    public void setRealUmsUserEnvironment(String realUmsUserEnvironmentKey) {
+        this.realUmsUserEnvironment = realUmsUserEnvironmentKey;
+    }
+
+    public void setInitializedKeys(Map<String, String> keys) {
+        this.initializedKeys = keys;
+    }
+
+    public void initUsers(String environmentKey, String accountKey) {
         String userConfigPath = "ums-users/api-credentials.json";
+        Map<String, Map<String, List<CloudbreakUser>>> fetchedUserRepository;
+        List<CloudbreakUser> cloudbreakUsers;
+        Set<String> accountIds = new HashSet<>();
+        String accountId;
+
+        LOGGER.info("Real UMS users are initializing by environment: {} and account: {}. Repository is present at: {} path", environmentKey, accountKey,
+                userConfigPath);
         try {
-            this.usersByAccount = JsonUtil.readValue(
-                    FileReaderUtils.readFileFromClasspathQuietly(userConfigPath), new TypeReference<>() {
-                    });
-        } catch (IOException e) {
-            throw new RuntimeException(String.format("Can't read file: %s It's possible you did run make fetch-secrets", userConfigPath));
+            fetchedUserRepository = JsonUtil.readValue(
+                    FileReaderUtils.readFileFromClasspathQuietly(userConfigPath), new TypeReference<Map<String, Map<String, List<CloudbreakUser>>>>() { });
```

Comment:
i would read json file during a PostConstruct method, and we need only a method without parameters which gives back users based on the environment and account which should be one during the test